modules/http/src/test/java/org/mule/module/http/functional/listener/HttpListenerAttachmentsTestCase.java

```diff
@@ -112,6 +122,47 @@ public void receiveOnlyAttachmentsAndReturnOnlyAttachmentsWithMultipartFormDataA
     }
 
     @Test
+    public void receiveAttachmentWithCustomHeaders() throws Exception
+    {
+
+        RequestBuilder requestBuilder = new RequestBuilder();
+        AsyncHttpClientConfig asyncHttpClientConfig = new AsyncHttpClientConfig.Builder().build();
+        GrizzlyAsyncHttpProvider grizzlyAsyncHttpProvider = new GrizzlyAsyncHttpProvider(asyncHttpClientConfig);
```

Comment:
Using fluent I couldn't customize the request and parse the multipart response as needed.